apps/dashboard/app/apps/user_preferences.rb

```diff
@@ -0,0 +1,43 @@
+class UserPreferences
+    require 'yaml'
+  
+    def initialize()
+      _prefs
+    end
+  
+    def preferences_file
+        @preferences_file ||= begin
+            file = ENV['OOD_PREFERENCES_FILE'] || "#{User.new.dir}/preferences.yml"
```

Comment:
Should indicate it's open ondemand related. how about `~/.config/ondemand/preferences.yml` (making the directory if it doesn't exist).